app/src/main/java/com/nevilleantony/prototype/adapters/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,58 @@
+package com.nevilleantony.prototype.adapters;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.nevilleantony.prototype.R;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewHolder> {
+
+    String data_name[],data_status[],data_speed[],data_percentage[];
+    Context context;
+    public RecyclerViewAdapter(Context ct, String s1[], String s2[], String s3[], String s4[]) {
+        context = ct;
+        data_name = s1;
+        data_status = s2;
+        data_speed = s3;
+        data_percentage = s4;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(context);
+        View view = inflater.inflate(R.layout.fragment_recyclerview_row,parent,false);
```

Comment:
As said above, let's avoid storing context references, so here we can use parent's context like so `LayoutInflater inflater = LayoutInflater.from(parent.getContext());`